### PR TITLE
Multisite template (Ema 0.8 upgrade)

### LINKED ISSUE #20: Improve this template
- [x] https://github.com/srid/ema/issues/75
- [ ] https://github.com/srid/ema/issues/76
- [ ] https://github.com/srid/ema/issues/79

### PR DESCRIPTION
**Pull request by srid**
Porting for https://github.com/srid/ema/pull/81 

Resolves #20 

- [x] Tailwind CSS cache issue in static site (live-server: no issue, due to use of UUID and CSS changing only on .hs change)

### CODE REVIEW

File: src/Main.hs [hunk 1]
```diff
@@ -80,195 +88,185 @@ markdownRouteFileBase :: MarkdownRoute -> Text
...
+      <> fmap Route_Static (allRoutes @StaticRoute $ snd . decomposeModel $ model)
+
+decomposeModel :: Model -> (Set MarkdownRoute, Set FilePath)
+decomposeModel = Map.keysSet . modelDocs &&& modelFiles
+
+routeDecomposition :: Iso' (Either MarkdownRoute StaticRoute) Route
+routeDecomposition =
+  iso (either Route_Markdown Route_Static) $ \case
+    Route_Markdown r -> Left r
+    Route_Static r -> Right r
```
Comment (srid) [inline src/Main.hs]: Too annoying to have the user do this. Ema should be able to generically derive this. https://github.com/srid/ema/pull/81#issuecomment-1163226891

Comment (srid) [reply src/Main.hs]: resolved in [f76280e](https://github.com/srid/ema-template/pull/22/commits/f76280e5189457a1ead3f2855bd03e62ca8cb8fc)

File: src/Main.hs [hunk 2]
```diff
@@ -37,16 +37,41 @@ import Text.Pandoc.Builder qualified as B
...
+    via ( Route
+            `WithSubRoutes` '[ SlugListRoute
+                             , StaticRoute
+                             ]
+        )
+  deriving (IsRoute) via (Route `WithModel` Model)
+
+instance HasSubModels Route where
+  subModels m =
+    I (Set.map coerce $ Map.keysSet $ modelDocs m) :* I (modelFiles m) :* Nil
```
Comment (srid) [inline src/Main.hs]: By the way, @lucasvreis - note here that the first sub-model (for markdown routes) is not really a sub-type (ie., it is not merely a field of the larger model) ... but rather an *arbitrary computation* (`Set.map coerce . Map.keysSet`) derived from that field.

We can conclude then that not all sub-models are sub-types.

Comment (lua-vr) [reply src/Main.hs]: Well, you could define `SlugMapRoute` on top of `SlugListRoute`, and this arbitary computation would then belong in `SlugMapRoute`

**Pull request merged**